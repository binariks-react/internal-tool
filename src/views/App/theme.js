export const getColor = (color, props) => (
  props ? props.theme.colors[color] : ({ theme }) => theme.colors[color]
);

export const getActiveColor = (color, props) => (
  props ? props.theme.colors.activeColors[color] : ({ theme }) => theme.colors.activeColors[color]
);

export const getHoverColor = (color, props) => (
  props ? props.theme.colors.hoverColors[color] : ({ theme }) => theme.colors.hoverColors[color]
);

export const getHoverShadowColor = (color, props) => (
  props ? props.theme.shadows.hover[color] : ({ theme }) => theme.shadows.hover[color]
);

export const getFocusShadowColor = (color, props) => (
  props ? props.theme.shadows.focus[color] : ({ theme }) => theme.shadows.focus[color]
);

export default {
  colors: {
    text: '#5a6169',
    lightGrey: '#e9ecef',
    darkGrey: '#becad6',
    border: '#e1e5eb',
    primary: '#007bff',
    secondary: '#5a6169',
    success: '#17c671',
    danger: '#c4183c',
    warning: '#ffb400',
    info: '#00b8d8',
    dark: '#212529',
    white: '#ffffff',
    royalBlue: '#674eec',
    java: '#1adba2',
    salmon: '#ff4169',
    hoverColors: {
      lightGrey: '#e9ecef',
      primary: '#006fe6',
      secondary: '#4e545b',
      success: '#14af64',
      danger: '#ad1535',
      warning: '#e6a200',
      info: '#00a2bf',
      dark: '#16181b',
      white: '#ffffff',
      royalBlue: '#5337ea',
      java: '#17c491',
      salmon: '#ff2855',
    },
    activeColors: {
      primary: '#0062cc',
      secondary: '#42484e',
      lightGrey: '#FBFBFB',
      success: '#129857',
      danger: '#97122e',
      warning: '#cc9000',
      info: '#008da5',
      dark: '#0a0c0d',
      white: '#007bff',
      royalBlue: '#3f20e7',
      java: '#15ad80',
      salmon: '#ff0e41',
    },
  },
  shadows: {
    hover: {
      primary: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(0,123,255,.25)',
      secondary: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(90,97,105,.25)',
      success: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(23,198,113,.25)',
      danger: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(196,24,60,.25)',
      warning: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(255,180,0,.25)',
      info: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(0,184,216,.25)',
      dark: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(33,37,41,.25)',
      white: '0 0.125rem 0.625rem rgba(129,142,163,.2), 0 0.0625rem 0.125rem rgba(129,142,164,.3)',
      royalBlue: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(103,78,236,.25)',
      java: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(26,219,162,.25)',
      salmon: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(255,65,105,.25)',
      inset: 'inset 0 3px 5px rgba(0,0,0,.125)!important',
    },
    focus: {
      primary: '0 0 0 3px rgba(0,123,255,.15),0 3px 15px rgba(0,123,255,.2),0 2px 5px rgba(0,0,0,.1)',
      secondary: '0 0 0 3px rgba(90,97,105,.15),0 3px 15px rgba(90,97,105,.2),0 2px 5px rgba(0,0,0,.1)',
      success: '0 0 0 3px rgba(23,198,113,.15),0 3px 15px rgba(23,198,113,.2),0 2px 5px rgba(0,0,0,.1)',
      danger: '0 0 0 3px rgba(196,24,60,.15),0 3px 15px rgba(196,24,60,.2),0 2px 5px rgba(0,0,0,.1)',
      warning: '0 0 0 3px rgba(255,180,0,.15),0 3px 15px rgba(255,180,0,.2),0 2px 5px rgba(0,0,0,.1)',
      info: '0 0 0 3px rgba(0,184,216,.15),0 3px 15px rgba(0,184,216,.2),0 2px 5px rgba(0,0,0,.1)',
      dark: '0 0 0 3px rgba(33,37,41,.15),0 3px 15px rgba(33,37,41,.2),0 2px 5px rgba(0,0,0,.1)',
      white: '0 0 0 3px rgba(251,251,251,.15),0 3px 15px rgba(251,251,251,.2),0 2px 5px rgba(0,0,0,.1)',
      royalBlue: '0 0 0 3px rgba(103,78,236,.15),0 3px 15px rgba(103,78,236,.2),0 2px 5px rgba(0,0,0,.1)',
      java: '0 0 0 3px rgba(26,219,162,.15),0 3px 15px rgba(26,219,162,.2),0 2px 5px rgba(0,0,0,.1)',
      salmon: '0 0 0 3px rgba(255,65,105,.15),0 3px 15px rgba(255,65,105,.2),0 2px 5px rgba(0,0,0,.1)',
      switch: '0 3px 8px 1px rgba(0,0,0,.2)',
    },
  },
};
