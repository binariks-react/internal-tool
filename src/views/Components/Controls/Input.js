import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const selectWrapperConditionStyle = ({ condition, theme: { colors: { success, danger } } }) => {
  switch (condition) {
  case null: return;
  case true: return `
                border-color: ${success};
                box-shadow: 0 5px 11.5px rgba(23,198,113,.1);
                `;
  case false: return `
                border-color: ${danger};
                box-shadow: 0 5px 11.5px rgba(196,24,60,.1);
                `;
  }
};

const messageWrapperConditionStyle = ({ condition, theme: { colors: { success, danger } } }) => {
  switch (condition) {
  case true: return `color: ${success}`;
  case false: return `color: ${danger}`;
  }
};

const Wrapper = styled.div`
  min-width: 7.5rem;
  width: 100%;
`;

const IconWrapper = styled.div`
  position: relative;
  display: flex;
  flex-wrap: nowrap;
  align-items: stretch;
  width: 100%;
  min-width: 7.5rem;  
`;

const InputWrapper = styled.input`
  padding: .4375rem .75rem;
  flex: 1 1 auto;
  font-size: .8125rem;
  line-height: 1.5;
  color: ${props => props.theme.colors.text};
  background-color: ${props => props.theme.colors.white};
  border: 1px solid ${props => props.theme.colors.border};
  font-weight: 300;
  border-radius: .25rem;
  box-shadow: none;
  transition: box-shadow 250ms cubic-bezier(.27,.01,.38,1.06), border 250ms cubic-bezier(.27,.01,.38,1.06);

  &:hover {
    border-color: ${props => props.theme.colors.hoverColors.border};
  }

  &:focus {
    color: ${props => props.theme.colors.text};
    background-color: ${props => props.theme.colors.white};
    border-color: ${props => props.theme.colors.primary};
    box-shadow: 0 0.313rem 0.719rem rgba(0,123,255,.1), 0 0.156rem 0.125rem rgba(0,0,0,.06);
  }

  ${props => selectWrapperConditionStyle(props)};

  ${({ icon }) =>
    icon && `
    border-top-left-radius: 0;
    border-bottom-left-radius: 0;
    `
};
`;

const MessageWrapper = styled.div`
  display: block;
  margin-top: .25rem;
  font-size: 80%;

  ${props => messageWrapperConditionStyle(props)};
`;

const FlexWrapper = styled.div`
  display: flex;
`;

const Icon = styled.span`
  font-size: .8125rem;
  padding: .375rem .75rem;
  font-weight: 300;
  line-height: 1.5;
  color: ${props => props.theme.colors.icon};
  background-color: ${props => props.theme.colors.white};
  border: 1px solid ${props => props.theme.colors.border};
  border-radius: .25rem;
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
  margin-right: -1px;
`;

const Input = props => (
  <Wrapper>
    <IconWrapper>
      {props.icon && <FlexWrapper><Icon>{props.icon}</Icon></FlexWrapper>}
      <InputWrapper
        {...props}
      />
    </IconWrapper>
    {props.message && <MessageWrapper {...props}>{props.message}</MessageWrapper>}
  </Wrapper>
);

Input.defaultProps = {
  icon: '',
  message: '',
  condition: null,
  placeholder: '',
  type: 'text',
};

Input.propTypes = {
  icon: PropTypes.string,
  message: PropTypes.string,
  condition: PropTypes.bool,
  placeholder: PropTypes.string,
  type: PropTypes.oneOf(['text', 'email', 'password']),
};

export default Input;
