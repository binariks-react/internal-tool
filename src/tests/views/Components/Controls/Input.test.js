import React from 'react';
import { ThemeProvider } from 'styled-components';
import { render } from 'enzyme';
import theme from 'views/App/theme';
import Input from 'views/Components/Controls/Input';
import 'jest-styled-components';

describe('Input', () => {
  describe('snapshot', () => {
    it('text input with placeholder', () => {
      const input = render(
        <ThemeProvider theme={theme}>
          <Input type="text" placeholder="Text" />
        </ThemeProvider>
      );
      expect(input).toMatchSnapshot();
    });

    it('password input with placeholder', () => {
      const input = render(
        <ThemeProvider theme={theme}>
          <Input type="password" placeholder="Password" />
        </ThemeProvider>
      );
      expect(input).toMatchSnapshot();
    });

    it('text input with placeholder and icon', () => {
      const input = render(
        <ThemeProvider theme={theme}>
          <Input type="text" placeholder="Input w/ icon" icon="@" />
        </ThemeProvider>
      );
      expect(input).toMatchSnapshot();
    });

    it('text input with placeholder and truthy condition', () => {
      const input = render(
        <ThemeProvider theme={theme}>
          <Input type="text" placeholder="Input w/ true condition" condition message="Condition is correct" />
        </ThemeProvider>
      );
      expect(input).toMatchSnapshot();
    });

    it('text input with placeholder and falsy condition', () => {
      const input = render(
        <ThemeProvider theme={theme}>
          <Input type="text" placeholder="Input w/ false condition" condition={false} message="Condition is incorrect" />
        </ThemeProvider>
      );
      expect(input).toMatchSnapshot();
    });
  });
});
