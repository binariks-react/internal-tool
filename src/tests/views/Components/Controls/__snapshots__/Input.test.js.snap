// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input snapshot password input with placeholder 1`] = `
.c0 {
  min-width: 7.5rem;
  width: 100%;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  width: 100%;
  min-width: 7.5rem;
}

.c2 {
  padding: .4375rem .75rem;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  font-size: .8125rem;
  line-height: 1.5;
  color: #5a6169;
  background-color: #ffffff;
  border: 1px solid #e1e5eb;
  font-weight: 300;
  border-radius: .25rem;
  box-shadow: none;
  -webkit-transition: box-shadow 250ms cubic-bezier(.27,.01,.38,1.06),border 250ms cubic-bezier(.27,.01,.38,1.06);
  transition: box-shadow 250ms cubic-bezier(.27,.01,.38,1.06),border 250ms cubic-bezier(.27,.01,.38,1.06);
}

.c2:hover {
  border-color: #b3bdcc;
}

.c2:focus {
  color: #5a6169;
  background-color: #ffffff;
  border-color: #007bff;
  box-shadow: 0 0.313rem 0.719rem rgba(0,123,255,.1),0 0.156rem 0.125rem rgba(0,0,0,.06);
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <input
      class="c2"
      placeholder="Password"
      type="password"
    />
  </div>
</div>
`;

exports[`Input snapshot text input with placeholder 1`] = `
.c0 {
  min-width: 7.5rem;
  width: 100%;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  width: 100%;
  min-width: 7.5rem;
}

.c2 {
  padding: .4375rem .75rem;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  font-size: .8125rem;
  line-height: 1.5;
  color: #5a6169;
  background-color: #ffffff;
  border: 1px solid #e1e5eb;
  font-weight: 300;
  border-radius: .25rem;
  box-shadow: none;
  -webkit-transition: box-shadow 250ms cubic-bezier(.27,.01,.38,1.06),border 250ms cubic-bezier(.27,.01,.38,1.06);
  transition: box-shadow 250ms cubic-bezier(.27,.01,.38,1.06),border 250ms cubic-bezier(.27,.01,.38,1.06);
}

.c2:hover {
  border-color: #b3bdcc;
}

.c2:focus {
  color: #5a6169;
  background-color: #ffffff;
  border-color: #007bff;
  box-shadow: 0 0.313rem 0.719rem rgba(0,123,255,.1),0 0.156rem 0.125rem rgba(0,0,0,.06);
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <input
      class="c2"
      placeholder="Text"
      type="text"
    />
  </div>
</div>
`;

exports[`Input snapshot text input with placeholder and falsy condition 1`] = `
.c0 {
  min-width: 7.5rem;
  width: 100%;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  width: 100%;
  min-width: 7.5rem;
}

.c2 {
  padding: .4375rem .75rem;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  font-size: .8125rem;
  line-height: 1.5;
  color: #5a6169;
  background-color: #ffffff;
  border: 1px solid #e1e5eb;
  font-weight: 300;
  border-radius: .25rem;
  box-shadow: none;
  -webkit-transition: box-shadow 250ms cubic-bezier(.27,.01,.38,1.06),border 250ms cubic-bezier(.27,.01,.38,1.06);
  transition: box-shadow 250ms cubic-bezier(.27,.01,.38,1.06),border 250ms cubic-bezier(.27,.01,.38,1.06);
  border-color: #c4183c;
  box-shadow: 0 5px 11.5px rgba(196,24,60,.1);
}

.c2:hover {
  border-color: #b3bdcc;
}

.c2:focus {
  color: #5a6169;
  background-color: #ffffff;
  border-color: #007bff;
  box-shadow: 0 0.313rem 0.719rem rgba(0,123,255,.1),0 0.156rem 0.125rem rgba(0,0,0,.06);
}

.c3 {
  display: block;
  margin-top: .25rem;
  font-size: 80%;
  color: #c4183c;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <input
      class="c2"
      placeholder="Input w/ false condition"
      type="text"
    />
  </div>
  <div
    class="c3"
    placeholder="Input w/ false condition"
    type="text"
  >
    Condition is incorrect
  </div>
</div>
`;

exports[`Input snapshot text input with placeholder and icon 1`] = `
.c0 {
  min-width: 7.5rem;
  width: 100%;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  width: 100%;
  min-width: 7.5rem;
}

.c4 {
  padding: .4375rem .75rem;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  font-size: .8125rem;
  line-height: 1.5;
  color: #5a6169;
  background-color: #ffffff;
  border: 1px solid #e1e5eb;
  font-weight: 300;
  border-radius: .25rem;
  box-shadow: none;
  -webkit-transition: box-shadow 250ms cubic-bezier(.27,.01,.38,1.06),border 250ms cubic-bezier(.27,.01,.38,1.06);
  transition: box-shadow 250ms cubic-bezier(.27,.01,.38,1.06),border 250ms cubic-bezier(.27,.01,.38,1.06);
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}

.c4:hover {
  border-color: #b3bdcc;
}

.c4:focus {
  color: #5a6169;
  background-color: #ffffff;
  border-color: #007bff;
  box-shadow: 0 0.313rem 0.719rem rgba(0,123,255,.1),0 0.156rem 0.125rem rgba(0,0,0,.06);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  font-size: .8125rem;
  padding: .375rem .75rem;
  font-weight: 300;
  line-height: 1.5;
  color: #abb6bf;
  background-color: #ffffff;
  border: 1px solid #e1e5eb;
  border-radius: .25rem;
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
  margin-right: -1px;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <span
        class="c3"
      >
        @
      </span>
    </div>
    <input
      class="c4"
      placeholder="Input w/ icon"
      type="text"
    />
  </div>
</div>
`;

exports[`Input snapshot text input with placeholder and truthy condition 1`] = `
.c0 {
  min-width: 7.5rem;
  width: 100%;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  width: 100%;
  min-width: 7.5rem;
}

.c2 {
  padding: .4375rem .75rem;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  font-size: .8125rem;
  line-height: 1.5;
  color: #5a6169;
  background-color: #ffffff;
  border: 1px solid #e1e5eb;
  font-weight: 300;
  border-radius: .25rem;
  box-shadow: none;
  -webkit-transition: box-shadow 250ms cubic-bezier(.27,.01,.38,1.06),border 250ms cubic-bezier(.27,.01,.38,1.06);
  transition: box-shadow 250ms cubic-bezier(.27,.01,.38,1.06),border 250ms cubic-bezier(.27,.01,.38,1.06);
  border-color: #17c671;
  box-shadow: 0 5px 11.5px rgba(23,198,113,.1);
}

.c2:hover {
  border-color: #b3bdcc;
}

.c2:focus {
  color: #5a6169;
  background-color: #ffffff;
  border-color: #007bff;
  box-shadow: 0 0.313rem 0.719rem rgba(0,123,255,.1),0 0.156rem 0.125rem rgba(0,0,0,.06);
}

.c3 {
  display: block;
  margin-top: .25rem;
  font-size: 80%;
  color: #17c671;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <input
      class="c2"
      placeholder="Input w/ true condition"
      type="text"
    />
  </div>
  <div
    class="c3"
    placeholder="Input w/ true condition"
    type="text"
  >
    Condition is correct
  </div>
</div>
`;
